### PR TITLE
Create Ctx object

### PR DESCRIPTION
**Pull request by izumin5210**

### CODE REVIEW

File: pkg/grapicmd/context.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+
+	"github.com/izumin5210/grapi/pkg/grapicmd/protoc"
+	"github.com/izumin5210/grapi/pkg/grapicmd/util/fs"
+)
+
+type Ctx struct {
```
Comment (watchbunny) [inline pkg/grapicmd/context.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type Ctx should have comment or be unexported

File: pkg/grapicmd/context.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	ProtocConfig protoc.Config
+}
+
+// Config stores general setting params and provides accessors for them.
+type Config struct {
+	ServerDir string
+	Package   string
+}
+
+func (c *Ctx) Init(cfgFile string) error {
```
Comment (watchbunny) [inline pkg/grapicmd/context.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Ctx.Init should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 3]
```diff
@@ -23,70 +23,70 @@ var (
 	gexCfgMu sync.Mutex
 )
 
-func ProvideUI(cfg *grapicmd.Config) clui.UI {
+func ProvideUI(ctx *grapicmd.Ctx) clui.UI {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideUI should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 4]
```diff
@@ -23,70 +23,70 @@ var (
...
 	if ui == nil {
-		ui = clui.New(cfg.OutWriter, cfg.InReader)
+		ui = clui.New(ctx.OutWriter, ctx.InReader)
 	}
 	return ui
 }
 
-func ProvideCommandExecutor(cfg *grapicmd.Config, ui clui.UI) command.Executor {
-	return command.NewExecutor(cfg.OutWriter, cfg.ErrWriter, cfg.InReader)
+func ProvideCommandExecutor(ctx *grapicmd.Ctx, ui clui.UI) command.Executor {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideCommandExecutor should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 5]
```diff
@@ -23,70 +23,70 @@ var (
...
+	return command.NewExecutor(ctx.OutWriter, ctx.ErrWriter, ctx.InReader)
 }
 
-func ProvideGenerator(cfg *grapicmd.Config, ui clui.UI) module.Generator {
+func ProvideGenerator(ctx *grapicmd.Ctx, ui clui.UI) module.Generator {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideGenerator should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 6]
```diff
@@ -23,70 +23,70 @@ var (
...
+		ctx.ProtocConfig.OutDir,
+		ctx.Config.ServerDir,
+		ctx.Config.Package,
+		ctx.Version,
 	)
 }
 
-func ProvideScriptLoader(cfg *grapicmd.Config, executor command.Executor) module.ScriptLoader {
-	return script.NewLoader(cfg.Fs, executor, cfg.RootDir)
+func ProvideScriptLoader(ctx *grapicmd.Ctx, executor command.Executor) module.ScriptLoader {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideScriptLoader should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 7]
```diff
@@ -23,70 +23,70 @@ var (
...
+	return script.NewLoader(ctx.FS, executor, ctx.RootDir)
 }
 
-func ProvideGexConfig(cfg *grapicmd.Config) *gex.Config {
+func ProvideGexConfig(ctx *grapicmd.Ctx) *gex.Config {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideGexConfig should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 8]
```diff
@@ -23,70 +23,70 @@ var (
...
+			WorkingDir: ctx.RootDir,
 			// TODO: set verbose flag
 			// TODO: set logger
 		}
 	}
 	return gexCfg
 }
 
-func ProvideInitializeProjectUsecase(cfg *grapicmd.Config, gexCfg *gex.Config, ui clui.UI, generator module.Generator) usecase.InitializeProjectUsecase {
+func ProvideInitializeProjectUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, generator module.Generator) usecase.InitializeProjectUsecase {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideInitializeProjectUsecase should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 9]
```diff
@@ -23,70 +23,70 @@ var (
...
 		ui,
 		generator,
 		gexCfg,
-		cfg.Version,
+		ctx.Version,
 	)
 }
 
-func ProvideExecuteProtocUsecase(cfg *grapicmd.Config, gexCfg *gex.Config, ui clui.UI, executor command.Executor, generator module.Generator) usecase.ExecuteProtocUsecase {
+func ProvideExecuteProtocUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, executor command.Executor, generator module.Generator) usecase.ExecuteProtocUsecase {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideExecuteProtocUsecase should have comment or be unexported

File: pkg/grapicmd/cmd/version.go [hunk 10]
```diff
@@ -9,20 +9,20 @@ import (
...
-			buf := bytes.NewBufferString(cfg.AppName + " " + cfg.Version)
-			if cfg.Prebuilt {
-				buf.WriteString(" (" + cfg.BuildDate + " " + cfg.Revision + ")")
+			buf := bytes.NewBufferString(ctx.AppName + " " + ctx.Version)
+			if ctx.Prebuilt {
+				buf.WriteString(" (" + ctx.BuildDate + " " + ctx.Revision + ")")
 			}
 			buf.WriteString("\n")
-			fmt.Fprintf(cfg.OutWriter, buf.String())
+			fmt.Fprintf(ctx.OutWriter, buf.String())
```
Comment (watchbunny) [inline pkg/grapicmd/cmd/version.go]: **[errcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	fmt.Fprintf(ctx.OutWriter, buf.String())

File: pkg/grapicmd/context.go (hunk 1 above)
Comment (izumin5210) [reply pkg/grapicmd/context.go]: fixed 👌

**Pull request merged**